### DISCUSSION TITLE
Does this mean we can have a Rust REPL?

### DISCUSSION THREAD

**Discussion by SichangHe** [FractalFir/rustc_codegen_clr#38] [Ideas]
If you can compile Rust to CLR, and CLR has REPLs, you can use Rust in a REPL?

**Comment by FractalFir** [comment]
Yes, and no. You will be able to use Rust _libraries_ in REPL, but you will not be able to use Rust in REPL _directly_.

**Reply by heavyrain266** [reply]
there is `rust-dev` component that exposes compiler in form of library which you can use to build your own REPL with `rustyline`.

**Reply by FractalFir** [reply]
Huh, I did not know about that! 

Still, my project will not enable people to use Rust *directly* in CLRs REPL.
